swarm/src/connection/handler_wrapper.rs

```diff
@@ -81,6 +77,15 @@ where
     shutdown: Shutdown,
     /// The substream upgrade protocol override, if any.
     substream_upgrade_protocol_override: Option<upgrade::Version>,
+    /// The maximum number of inbound streams concurrently negotiating. New inbound
+    /// streams exceeding the limit are dropped and thus reset.
+    ///
+    /// Note: This only enforces a limit on the number of concurrently
+    /// negotiating inbound streams. The total number of inbound streams on a
+    /// connection is the sum of negotiating and negotiated streams. A limit on
+    /// the total number of streams can be enforced at the [`StreamMuxerBox`]
+    /// level.
+    max_number_negotiating_inbound_streams: usize,
```

Comment:
Nit: `number` might be redundant here.

Reply:
:+1: Renamed with https://github.com/libp2p/rust-libp2p/pull/2697/commits/2b187af6507f29a552e3dfd0eff026aee3fd3317.